File: bin/cmds/telnet.pl
#!/usr/bin/env perl

use strict;
use warnings;

my $host=shift(@ARGV);
my $port=shift(@ARGV);
my $user=shift(@ARGV);
my $password=shift(@ARGV);
my $prompt=shift(@ARGV);
my $cmd=join(" ",@ARGV);

open(P,"telnet.exp $host $port $user $password $prompt '$cmd' |") or die;
my $l=0;
my $pr=0;
my $s="";
while (<P>) {
	s/\r//g;
	++$l;
	if (($l==1) && /^spawn/) {
		$s.=$_;
		next;
	}
	if ($pr==0 && /$prompt/) {
		$pr=1;
		$s.=$_;
		next;
	}
	if ($pr==0) {
		$s.=$_;
		next;
	}
	if ($pr==1 && /$prompt/) {
		$pr=0;
		$s.=$_;
		next;
	}
	if (/\+\+\+\-exit\-\+\+\+ (\d+)/) {
		print STDERR "exit $1\n";
		exit $1;
	}
	print;
}
if (!close(P)) {
	print STDERR "$s";
	exit 2;
}

File: bin/cmds/telnet.py
#!/usr/bin/env python

import os
import sys
import re

def main():

    prompt = sys.argv[5]
    args = " "
    for s in sys.argv[1:]:
        args += ('\'' + s + '\' ')

    pt = False
    l = 0
    s = ""
    exitcode = 0

    r = os.popen("telnet.exp" + args)
    lines = r.readlines()
    for line in lines:
        l += 1
        line = line.replace('\r\n', '\n')
        if l == 1 and re.search('^spawn', line):
            s += line
            continue

        if pt is False and re.search(prompt, line):
            pt = True
            s += line
            continue

        if pt is False:
            s += line
            continue

        if pt is True and re.search(prompt, line):
            s += line
            continue

        if re.search('\+\+\+\-exit\-\+\+\+ (\d+)', line):
            s += line
            exitcode = 1
            break

        sys.stdout.write(line)

    r.close()

    if pt is False:
        exitcode = 2

    if exitcode:
        sys.stderr.write(s)
        sys.stderr.write('exit ' + str(exitcode) + '\n')

    sys.exit(exitcode)

if __name__ == '__main__':
    main()